sdks/python/apache_beam/io/gcp/gcsio.py

```diff
@@ -156,10 +157,10 @@ class GcsIOError(IOError, retry.PermanentException):
 
 class GcsIO(object):
   """Google Cloud Storage I/O client."""
-  def __init__(self, storage_client=None):
+  def __init__(self, storage_client=None, pipeline_options={}): # pylint: disable=dangerous-default-value
     if storage_client is None:
       storage_client = storage.StorageV1(
-          credentials=auth.get_service_credentials(),
+          credentials=auth.get_service_credentials(pipeline_options),
```

Comment:
Will credentials singleton ever be created here during job submission path? Does it make sense to define `def get_service_credentials(pipeline_options=None):` so that you only need to pass the options in the runners (DataflowRunner, Interactive Runner, etc), but don't have to modify the IO code that instantiates an already created credentials, or executes after job submission?